cli/azd/pkg/osutil/rename_windows.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//go:build windows
+// +build windows
+
+package osutil
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+	"golang.org/x/sys/windows"
+)
+
+// Rename is like os.Rename except it will retry the operation, up to 10 times, waiting a second between each retry when the
+// Rename fails due to what may be transient file system errors. This can help work around issues where the file may
+// temporary be opened by a virus scanner or some other process which prevents us from renaming the file.
+func Rename(ctx context.Context, old, new string) error {
+	return retry.Do(ctx, retry.WithMaxRetries(10, retry.NewConstant(1*time.Second)), func(ctx context.Context) error {
```

Comment:
Initially I had this as a 5 second pause with a half a second backoff as you described.  I realized that for the `ERROR_ACCESS_DENIED` case, this would happen when the target file (bicep.exe) was in use because it was executing.  I was a little nervous about bicep completing in 5 seconds, so I bumped the overall time (and increased the delay).